src/EFCore.Jet/Query/ExpressionTranslators/Internal/JetStringMethodTranslator.cs

```diff
@@ -111,7 +111,7 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
             // _trimWithNoParam is only available since .NET Core 2.0 (or .NET Standard 2.1).
             if (Equals(method, _trimWithNoParam) ||
                 Equals(method, _trimWithChars) &&
-                (arguments[0] as SqlConstantExpression)?.Value == null || ((arguments[0] as SqlConstantExpression)?.Value as Array)?.Length == 0)
+                (!arguments.Any() || (arguments[0] as SqlConstantExpression)?.Value == null || ((arguments[0] as SqlConstantExpression)?.Value as Array)?.Length == 0))
```

Comment:
Actually, it seems the original code is just missing parenthesis around the second && argument:

```c#
if (Equals(method, _trimWithNoParam) ||
    Equals(method, _trimWithChars) && ((arguments[0] as SqlConstantExpression)?.Value == null || ((arguments[0] as SqlConstantExpression)?.Value as Array)?.Length == 0))
```

The same is true for `LTRIM` and `RTRIM`.

Reply:
Oh yeah, missed that! Pushed your suggested fix.